app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala

```diff
@@ -42,17 +42,20 @@ abstract class PrivilegedCitizenMatchingController(
 
   def matchCitizen = Action.async(BodyParsers.parse.json) { implicit request =>
     requiresPrivilegedAuthentication(scopeService.getAllScopes) { _ =>
-      withJsonBody[CitizenMatchingRequest] { matchCitizen =>
-        citizenMatchingService.matchCitizen(matchCitizen) map { matchId =>
-          val selfLink = HalLink("self", s"/individuals/matching/")
-          val individualLink = HalLink(
-            "individual",
-            s"/individuals/matching/$matchId",
-            name = Option("GET"),
-            title = Option("Individual Details"))
-          Ok(links(selfLink, individualLink))
+      withCorrelationId { () =>
```

Comment:
I don't like the use of blank parameters... is it possible to define it without?

Reply:
i.e. withCorrelationId {
 withJsonBody { matchCitizen => ...
 }
}

Reply:
This doesnt work. Unsure why, might be as theres no paramter required in the function other than the implicit, Might just be my misunderstanding of scala syntax.
Required : Function0[Future[Result]]
Found: Future[Result] 
when trying the above.